### DISCUSSION TITLE
Cannot push image to ghcr via workflow but cli works

### DISCUSSION THREAD

**Discussion by aptalca** [community/community#26752] [Packages]
<p>Hi,</p>
<p>I’m using a bot account to push images to ghcr.</p>
<p>When I use the pat created for the bot, I can log into ghcr and then successfully push image manually in cli.</p>
<p>But if I try to use the same user and pat in a github workflow, it fails with <code>denied: permission_denied: write_package</code></p>
<p>In org settings, users are allowed to push packages as <code>public</code>, <code>private</code> and <code>internal</code> (<a class="mention" href="/clarkbw">@clarkbw</a> had turned on the internal flag for our org). And the pat has the read and write package perms enabled.</p>
<p>With the same user/pat, cli push works just fine and the new package is marked <code>internal</code> as expected. But a push from workflow with the same credentials fails: <a href="https://github.com/linuxserver/docker-mods/runs/1338930900?check_suite_focus=true" rel="noopener nofollow ugc">https://github.com/linuxserver/docker-mods/runs/1338930900?check_suite_focus=true</a></p>
<p>Any ideas?</p>
<p>Thanks</p>

**Comment by aptalca** [comment, marked as answer]
<p>It turns out another org member had pushed the same package, which was private by default and was owned by that org member.</p>
<p>Since nobody else could even see the package as existing, we were very confused.</p>
<p>I think this default behavior of new packages being privately owned by the user uploading and not being visible to even the org owners is quite confusing.</p>

**Comment by whitneyimura** [comment]
<aside class="quote no-group" data-username="aptalca" data-post="2" data-topic="141055">
<div class="title">
<div class="quote-controls"></div>
 aptalca:</div>
<blockquote>
<p>I think this default behavior of new packages being privately owned by the user uploading and not being visible to even the org owners is quite confusing.</p>
</blockquote>
</aside>
<p>:wave: Hi there – Yes, we’ve received this feedback and are working on a fix to make this more intuitive for the end user.  :heart:</p>

**Comment by nemchik** [comment]
<aside class="quote no-group" data-username="aptalca" data-post="2" data-topic="141055" data-full="true">
<div class="title">
<div class="quote-controls"></div>
 aptalca:</div>
<blockquote>
<p>It turns out another org member had pushed the same package, which was private by default and was owned by that org member.</p>
</blockquote>
</aside>
<p><img src="https://user-images.githubusercontent.com/725456/181099759-c0fa1796-1fd4-4904-9101-e9bfa74c76b0.gif" alt="" data-base62-sha1="hFB30tV29vA7CeKVj8mIwog19FX" width="220" height="220"></p>

**Comment by halink0803** [comment]
<p>I also got the same problem but mine is that I deleted the old repository and create a new one (with the same name). I cannot push the packages. I go to package tab in my profile and link the old package to the new repo but still cannot push even though I own both of them. I think you guys should fix it. Simple solution is change the target to push package to different name.</p>

**Comment by danielhelfand** [comment]
<p>I have also noticed this issue with deleting a repository and then recreating the repository with the same name. This is really relevant in the context of forking a repository. I personally debug/test my project’s release process using a fork of the project. This makes it very challenging for our current workflow if we need to rename the repository. Agreed there should be some kind of fix for this.</p>

**Comment by ulidtko** [comment]
<p>I’m affected by the same issue. The token evidently has the <code>write:packages</code> permission — but the push error is extremely confusing, claiming a misleading <code>denied: permission_denied: write_package</code>.</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/9/c/9c24121a7e2623fc34aaae17437a38b7ab24f8a1.png" data-download-href="/uploads/short-url/mhhLwAXEfJSqBDro03xfKFZk5z3.png?dl=1" title="image" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/365338/181099761-f27c4190-e52d-416a-8395-0d93c39a8b6d.png" alt="image" data-base62-sha1="mhhLwAXEfJSqBDro03xfKFZk5z3" width="335" height="500" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/9/c/9c24121a7e2623fc34aaae17437a38b7ab24f8a1_2_335x500.png, https://aws1.discourse-cdn.com/github/optimized/3X/9/c/9c24121a7e2623fc34aaae17437a38b7ab24f8a1_2_502x750.png 1.5x, https://aws1.discourse-cdn.com/github/original/3X/9/c/9c24121a7e2623fc34aaae17437a38b7ab24f8a1.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/9/c/9c24121a7e2623fc34aaae17437a38b7ab24f8a1_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">image</span><span class="informations">548×817 51.4 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>In my case, the workflow tries to publish to an <code>Internal</code> org-level Package (i.e. <code>ghcr.io/PrivateOrg/haskell-stack-builder:tag</code>). Neither updating an existing tag, nor pushing a new one works from Actions.</p>
<p>Pushing manually from CLI works OK.</p>
<aside class="quote no-group" data-username="halink0803" data-post="5" data-topic="141055">
<div class="title">
<div class="quote-controls"></div>
 halink0803:</div>
<blockquote>
<p>Simple solution is change the target to push package to different name.</p>
</blockquote>
</aside>
<p>The “simplicity” of this solution vanishes real quick once you start having dozens of other Workflows using this Package.</p>

**Comment by ulidtko** [comment]
<p>From a user perspective, looks insistently like a bug in the permission system. For example, I keep getting this “permission denied” — even after gaining <strong>package admin</strong> role.</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/b/2/b290bb48e723ca4d9f31c867dc89bbdd9153697f.png" data-download-href="/uploads/short-url/ptF4H4MApBJyrO6Jm21Rj0TMSTB.png?dl=1" title="image" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/365338/181099762-0c46c7b7-09c3-43ea-8466-55273269c562.png" alt="image" data-base62-sha1="ptF4H4MApBJyrO6Jm21Rj0TMSTB" width="690" height="159" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/b/2/b290bb48e723ca4d9f31c867dc89bbdd9153697f_2_690x159.png, https://aws1.discourse-cdn.com/github/original/3X/b/2/b290bb48e723ca4d9f31c867dc89bbdd9153697f.png 1.5x, https://aws1.discourse-cdn.com/github/original/3X/b/2/b290bb48e723ca4d9f31c867dc89bbdd9153697f.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/b/2/b290bb48e723ca4d9f31c867dc89bbdd9153697f_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">image</span><span class="informations">1005×232 12.1 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>It’s really really really easy to get confused.</p>
<p>In the end, I got it working with the help of our org admin (and, like, 15 test rebuilds).</p>
<p>What helped was this URL (only admin can open it):<br>
<code>https://github.com/orgs/PrivateOrg/packages/container/haskell-stack-builder/settings</code><br>
— and then, <em>granting the <code>write:packages</code> permission <strong>to the repo</strong></em> where the publishing Workflow runs — in my case, <code>github.com/PrivateOrg/haskell-stack-builder.git</code>:</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/1/c/1c2ba141d363110d01c8d54c683d4ec7804d307a.png" data-download-href="/uploads/short-url/41cQrlNwuBAt1o7hPfiIkedWXyi.png?dl=1" title="image" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/365338/181099763-8d9b4e1a-9948-498a-bf78-07aa96213cc0.png" alt="image" data-base62-sha1="41cQrlNwuBAt1o7hPfiIkedWXyi" width="690" height="78" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/1/c/1c2ba141d363110d01c8d54c683d4ec7804d307a_2_690x78.png, https://aws1.discourse-cdn.com/github/optimized/3X/1/c/1c2ba141d363110d01c8d54c683d4ec7804d307a_2_1035x117.png 1.5x, https://aws1.discourse-cdn.com/github/original/3X/1/c/1c2ba141d363110d01c8d54c683d4ec7804d307a.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/1/c/1c2ba141d363110d01c8d54c683d4ec7804d307a_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">image</span><span class="informations">1077×122 5.2 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>This has resolved the error for me.</p>
